src/early-types.lisp

```diff
@@ -53,17 +53,67 @@
 (defvar *type-parsers* (make-hash-table :test 'equal)
   "Hash table of defined type parsers.")
 
+(define-condition cffi-error (error)
+  ())
+
+(define-condition type-related-error (cffi-error)
```

Comment:
yeah, i'm not happy with that either, but it's still better, so i'll rename them. this one is just a mixin, but this naming pattern is even worse in the later condition classes.

shall i get rid of the class-name/slot accessors?